fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -83,7 +83,13 @@
             "Search parameter: '%s' must have resource type name modifier";
     private static final String INVALID_TARGET_TYPE_EXCEPTION = "Invalid target type for the Inclusion Parameter.";
     private static final String UNSUPPOTED_EXPR_NULL =
-            "An empty expression is found or the parameter type is unsupported [%s][%s]";;
+            "An empty expression is found or the parameter type is unsupported [%s][%s]";
+    
+    private static final String MODIFIYERRESOURCETYPE_NOT_ALLOWED_FOR_RESOURCETYPE =
```

Comment:
I think Modifier is not allowed for search parameter of resource type would be sufficient.

Reply:
I think we'd better make the error message more accurate.